modules/fb_instant_articles_display/fb_instant_articles_display.module

```diff
@@ -0,0 +1,777 @@
+    ->execute();
+  // Allow other modules to perform actions.
+  module_invoke_all('fb_instant_articles_display_set_type', $type, $bundle);
+}
+
+/**
+ * Delete the field set to be used as the product id for entity.
+ *
+ * @param string $type
+ *   The entity type.
+ * @param string $bundle
+ *   The entity bundle.
+ */
+function fb_instant_articles_display_delete_entity_type($type, $bundle) {
+  db_delete('fb_instant_articles_display_entity_types')
+    ->condition('entity_type', $type)
+    ->condition('entity_bundle', $bundle)
+    ->execute();
+  // Allow other modules to perform actions.
+  module_invoke_all('fb_instant_articles_display_delete_type', $type, $bundle);
+}
+
+/**
+ * Implements hook_entity_info_alter().
+ */
+function fb_instant_articles_display_entity_info_alter(&$entity_info) {
+  $entity_info['node']['view modes']['fb_instant_article'] = array(
+    'label' => t('Facebook Instant Articles'),
+    'custom settings' => TRUE,
+  );
+}/**
+ * Implements hook_preprocess_node().
+ */
+function fb_instant_articles_display_preprocess_node(&$vars) {
+
+  if ($vars['view_mode'] == 'fb_instant_article') {
+
+    $layout_settings = fb_instant_articles_display_get_node_layout_settings($vars['type']);
+
+    // If no regions are set, return early.
+    if (empty($layout_settings->settings['regions'])) {
+      return;
+    }
+
+    $regions = $layout_settings->settings['regions'];
+
+    foreach ($regions as $region => $fields) {
+      foreach ($fields as $field) {
+
+        if (!isset($vars['content'][$field])) {
+          continue;
+        }
+
+        $field_array = $vars['content'][$field];
+
+        // TODO: Make title field format configurable.
+        if ($field == 'title') {
+          $field_array = array(
+            '#markup' => '<h1>' . $vars['title'] . '</h1>',
+          );
+        }
+
+        $vars['fb_instant_article_' . $region][$field] = $field_array;
+      }
+    }
+
+    // Set the style of the article from configuration.
+    $vars['fb_instant_article_style'] = variable_get('fb_instant_articles_display_style', 'default');
+
+    $vars['canonical_url'] = url('node/' . $vars['nid'], array('absolute' => TRUE));
+    $vars['published'] = fb_instant_articles_display_time_element($vars['created'], 'op-published');
+    $vars['modified'] = fb_instant_articles_display_time_element($vars['changed'], 'op-modified');
+
+    $vars['theme_hook_suggestions'][] = 'node__fb_instant_article';
+  }
+}
+
+/**
+ * Implements hook_preprocess_node().
+ */
+function fb_instant_articles_display_preprocess_field(&$vars) {
+  $element = $vars['element'];
+  if ($element['#view_mode'] == 'fb_instant_article') {
+    $vars['theme_hook_suggestions'][] = 'field__fb_instant_article';
+  }
+}
+
+/**
+ * Implements hook_theme().
+ */
+function fb_instant_articles_display_theme() {
+  $theme_path = drupal_get_path('module', 'fb_instant_articles_display') . '/theme';
+  return array(
+    'node__fb_instant_articles_display' => array(
```

Comment:
No need for rename, should be `node__fb_instant_article`.  This theme hook is designed to override the template suggestion for the nodes rendered as `fb_instant_article` view mode